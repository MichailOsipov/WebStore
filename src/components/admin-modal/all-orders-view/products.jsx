import * as React from 'react';
import {ProductCard} from 'components/product-card';

export const Products = ({products}) => (
    <div>
        {products.map(({id, name, imageSrc}) => (
            <ProductCard
                key={id}
                name={name}
                imageSrc={imageSrc}
            />
        ))}
    </div>
);
