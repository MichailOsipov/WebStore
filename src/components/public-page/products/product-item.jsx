import * as React from 'react';
import {Button} from '@material-ui/core';

export const ProductItem = ({id, name, onOpenProductInformation}) => (
    <div key={id}>
        {name}
        <Button variant="contained" onClick={onOpenProductInformation}>Подробнее</Button>
    </div>
);
