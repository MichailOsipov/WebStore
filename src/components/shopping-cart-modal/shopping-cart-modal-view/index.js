export {ShoppingCartModalView} from './shopping-cart-modal-view';
