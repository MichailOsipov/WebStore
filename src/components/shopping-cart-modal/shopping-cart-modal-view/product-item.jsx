import * as React from 'react';
import Button from '@material-ui/core/Button';

export const ProductItem = ({name, onRemoveItem}) => (
    <div>
        {name}
        <Button variant="contained" onClick={onRemoveItem}>Удалить</Button>
    </div>
);
