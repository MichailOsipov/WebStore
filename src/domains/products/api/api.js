import {MOCK_PRODUCTS} from './mock-products';
import {MOCK_PRODUCT} from './mock-product';
import {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
import {MOCK_ORDERS} from './mock-orders';

export const ProductsApi = {
    loadProducts: () => Promise.resolve(MOCK_PRODUCTS),
    loadProduct: productId => Promise.resolve(MOCK_PRODUCT),
    addProductToCart: productId => Promise.resolve(),
    loadProductsAddedToCart: () => Promise.resolve(MOCK_PRODUCTS_ADDED_TO_CART),
    removeProductFromCart: productId => Promise.resolve(),
    makeAnOrder: () => Promise.resolve(),
    loadOrders: () => Promise.resolve(MOCK_ORDERS)
};
