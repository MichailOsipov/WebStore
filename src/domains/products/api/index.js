export {ProductsApi} from './api';
export {MOCK_ALL_ORDERS} from './mock-all-orders';
export {MOCK_ORDERS} from './mock-orders';
export {MOCK_PRODUCT} from './mock-product';
export {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
export {MOCK_PRODUCTS} from './mock-products';
