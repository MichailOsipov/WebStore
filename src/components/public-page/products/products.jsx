import * as React from 'react';
import {connect} from 'react-redux';
import {Grid, Typography} from '@material-ui/core';
import {getProducts, fetchProducts} from 'domains/products/state';
import {ProductItem} from './product-item';

export const Products = connect(
    state => ({
        products: getProducts(state)
    }),
    {
        fetchProducts
    }
)(
    class extends React.Component {
        componentDidMount() {
            this.props.fetchProducts();
        }

        render() {
            const {onOpenProductInformation, products} = this.props;
            return (
                <Grid container justify="center">
                    <Grid item>
                        <Typography variant="h6">Товары в наличии:</Typography>
                    </Grid>
                    <Grid container justify="center">
                        {products.map(({id, name, imageSrc}) => (
                            <Grid item xs={2} key={id}>
                                <ProductItem
                                    name={name}
                                    imageSrc={imageSrc}
                                    onOpenProductInformation={() => onOpenProductInformation(id)}
                                />
                            </Grid>
                        ))}
                    </Grid>
                </Grid>
            );
        }
    });
