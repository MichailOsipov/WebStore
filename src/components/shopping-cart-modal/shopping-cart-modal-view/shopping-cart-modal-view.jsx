import * as React from 'react';
import {Button, Typography} from '@material-ui/core';
import {ProductItem} from './product-item';

export const ShoppingCartModalView = ({
    makeAnOrder,
    productsAddedToCart,
    removeProductFromCart,
    onGoMainPage
}) => (
    <div>
        <Typography variant="h6">Корзина</Typography>
        {productsAddedToCart.map(({id, name, imageSrc}) => (
            <ProductItem
                key={id}
                name={name}
                imageSrc={imageSrc}
                onRemoveItem={() => removeProductFromCart(id)}
            />
        ))}
        <Button variant="contained" onClick={makeAnOrder}>Сделать заказ</Button>
        <Button onClick={onGoMainPage}>На главную</Button>
    </div>
);
