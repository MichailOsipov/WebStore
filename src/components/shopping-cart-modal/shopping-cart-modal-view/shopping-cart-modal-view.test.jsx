import * as React from 'react';
import {shallow} from 'enzyme';
import {MOCK_PRODUCTS_ADDED_TO_CART} from 'domains/products/api';
import {ShoppingCartModalView} from './shopping-cart-modal-view';

describe('ShoppingCartModalView', () => (
    test('rendering', () => {
        const wrapper = shallow(
            <ShoppingCartModalView
                makeAnOrder={jest.fn()}
                productsAddedToCart={MOCK_PRODUCTS_ADDED_TO_CART}
                removeProductFromCart={jest.fn()}
                onGoMainPage={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    })
));
