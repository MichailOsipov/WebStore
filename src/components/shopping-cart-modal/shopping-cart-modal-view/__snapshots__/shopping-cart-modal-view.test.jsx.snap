// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCartModalView rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="column"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Typography)
      variant="h5"
    >
      Корзина
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      container={true}
      spacing={16}
    >
      <WithStyles(Grid)
        item={true}
        key="3"
        xs={3}
      >
        <ProductItem
          imageSrc="https://www.stroydomtrust.ru/assets/gallery/1191/8076.jpg"
          name="Дом"
          onRemoveItem={[Function]}
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="4"
        xs={3}
      >
        <ProductItem
          imageSrc="https://static-eu.insales.ru/images/products/1/1289/73352457/large_INES_main.jpg"
          name="Стол"
          onRemoveItem={[Function]}
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Button)
      onClick={[MockFunction]}
      variant="contained"
    >
      Сделать заказ
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      На главную
    </WithStyles(Button)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
